evm_arithmetization/src/cpu/syscalls_exceptions.rs

```diff
@@ -34,16 +34,18 @@ pub(crate) fn eval_packed<P: PackedField>(
     yield_constr.constraint(filter_syscall * (filter_syscall - P::ONES));
     yield_constr.constraint(filter_exception * (filter_exception - P::ONES));
 
-    // If exception, ensure we are not in kernel mode
-    yield_constr.constraint(filter_exception * lv.is_kernel_mode);
-
     // Get the exception code as an value in {0, ..., 7}.
     let exc_code_bits = lv.general.exception().exc_code_bits;
     let exc_code: P = exc_code_bits
         .into_iter()
         .enumerate()
         .map(|(i, bit)| bit * P::Scalar::from_canonical_u64(1 << i))
         .sum();
+
+    // If exception but not final halting step, ensure we are not in kernel mode.
```

Comment:
> If exception but not final halting step

reads a bit weird
 + needs explanation why we're skipping one value (the `exc_stop` for segmentation)

Reply:
(same for identical comment in recursive mode)